internal/jujuapi/jimm_unit_test.go

```diff
@@ -80,3 +105,48 @@ func (s *jimmSuite) TestPrepareModelMigration_InvalidUserMapping(c *gc.C) {
 
 	c.Assert(err, gc.ErrorMatches, `--badwolf--@canonical.com is not a valid external user name`)
 }
+
+func (s *jimmSuite) TestBootstrapStatus(c *gc.C) {
+	ctx := context.Background()
+	uuidGenerated := uuid.New()
+	jimm := &jimmtest.JIMM{
+		BootstapManager_: func() jimm.BootstrapManager {
+			return &mocks.BootstapManager{
+				GetBootstrapStatusAndLogs_: func(ctx context.Context, user *openfga.User, jobId uuid.UUID, offset int) (params.BootstrapStatusResponse, error) {
+					if jobId != uuidGenerated {
+						return params.BootstrapStatusResponse{}, errors.E(errors.CodeNotFound, "job not found")
+					}
+					return params.BootstrapStatusResponse{
+						Status: "running",
+						Logs:   []string{"bootstrap logs"},
+					}, nil
+				},
+			}
+		},
+	}
+	root := newTestControllerRoot(jimm, "alice@canonical.com", true)
+
+	response, err := root.BootstrapStatus(ctx, params.BootstrapStatusRequest{
+		JobID:     uuidGenerated.String(),
+		Watermark: 0,
+	})
+
+	c.Assert(err, gc.IsNil)
+	c.Assert(response.Status, gc.Equals, "running")
+	c.Assert(response.Logs, gc.DeepEquals, []string{"bootstrap logs"})
+
+	// Test job not found
+	_, err = root.BootstrapStatus(ctx, params.BootstrapStatusRequest{
+		JobID:     uuid.New().String(),
+		Watermark: 0,
+	})
+	c.Assert(errors.ErrorCode(err), gc.Equals, errors.CodeNotFound)
+
+	// Test unauthorized user
+	root = newTestControllerRoot(jimm, "alice@canonical.com", false)
```

Comment:
Yeah as I read this it is feeling like it doesn't make sense that an admin can view the progress of a bootstrap that might not be theirs.